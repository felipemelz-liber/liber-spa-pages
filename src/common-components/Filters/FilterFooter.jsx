import React from 'react';
import PropTypes from 'prop-types';
import { Footer, FilterButton, ClearButton } from './Filters.styles';

const FilterFooter = ({ disabled, onFilter, onClear }) => {
  return (
    <Footer>
      <FilterButton disabled={disabled} onClick={onFilter}>
        FILTRAR
      </FilterButton>
      <ClearButton onClick={onClear} disabled={disabled}>
        LIMPAR
      </ClearButton>
    </Footer>
  );
};

FilterFooter.propTypes = {
  disabled: PropTypes.bool,
  onFilter: PropTypes.func,
  onClear: PropTypes.func,
};

FilterFooter.defaultProps = {
  disabled: false,
  onFilter: () => null,
  onClear: () => null,
};

export default FilterFooter;
