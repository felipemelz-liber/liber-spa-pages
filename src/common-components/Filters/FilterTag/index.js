export { default } from './FilterTag.jsx';
