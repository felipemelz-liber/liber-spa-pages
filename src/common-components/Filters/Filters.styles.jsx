import styled, { css } from 'styled-components';
import { ModalTag, CardFooter, Button, handleThemeFromObject, TextField } from 'liber-components';

const getTheme = (key, fallback) =>
  css`
    ${({ theme }) => handleThemeFromObject(theme, key, fallback)}
  `;

export const Container = styled.div`
  min-height: 68px;
  width: 100%;
  border-right: 1px solid #cdd7df;
  border-left: 1px solid #cdd7df;
  padding: 0px 16px;
  display: flex;
  justify-content: flex-start;
  align-items: center;
`;

export const GreyFilterTag = styled(ModalTag)`
  font-weight: 400;
`;

export const Label = styled.div`
  font-family: Roboto;
  font-size: 16px;
  line-height: 22px;
  letter-spacing: 0.1px;
  text-align: left;
  color: #405f71;
  margin-right: 16px;
`;

export const Footer = styled(CardFooter)`
  width: 100%;
  min-height: 52px;
  height: 52px;
  background-color: #ffffff;
  border-radius: 0px 0px 4px 4px;
  padding-top: unset;
  padding-bottom: unset;
  display: flex;
  justify-content: flex-start;
  align-items: center;
  flex-direction: row-reverse;
`;

export const ClearButton = styled(Button).attrs(() => ({ version: '2' }))`
  font-weight: 400;
`;

export const FilterButton = styled(Button).attrs(() => ({ version: '2' }))`
  font-weight: 400;
  color: ${getTheme('colors.liber', '#009dff')};
`;

export const Field = styled(TextField)``;

export const CleanAll = styled(FilterButton)`
  font-size: 16px;
  line-height: 22px;
  letter-spacing: 0.1px;
`;

export const Filtered = styled.div`
  font-family: Roboto;
  font-size: 16px;
  font-weight: 500;
  line-height: 22px;
  letter-spacing: 0.1px;
  text-align: right;
  color: #405f71;
  margin-left: auto;
`;
