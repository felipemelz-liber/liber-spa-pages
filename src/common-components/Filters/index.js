export { default as ArrayFieldFilter } from './ArrayFieldFilter/ArrayFieldFilter.jsx';
export { default as TextFieldFilter } from './TextFieldFilter/TextFieldFilter.jsx';
export { default as DateFieldFilter } from './DateFieldFilter/DateFieldFilter.jsx';
