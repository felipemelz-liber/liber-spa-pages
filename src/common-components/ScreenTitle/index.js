export { default } from './ScreenTitle.jsx';
