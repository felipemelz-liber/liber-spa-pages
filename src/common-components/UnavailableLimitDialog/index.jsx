import React from 'react';
import PropTypes from 'prop-types';
import { Modal } from 'liber-components';

import { Container, Header, Content, ActionsRow, Button } from './styles';
import { formatMoney } from '../../views/withdraw/moneyHandler';

function UnavailableLimitDialog({ surplusValue, onLeaved, show, redirectUrl }) {
  const handleOk = redirectUrl ? () => window.location.assign(redirectUrl) : onLeaved;

  return (
    <Modal onLeaved={onLeaved} show={show}>
      <Container>
        <Header>Limite Indisponível</Header>
        <Content>
          {surplusValue ? (
            <>
              A antecipação não poderá ser realizada, pois o valor líquido excedeu o seu limite
              disponível.
              <br />
              <br />O excesso tem o valor de {formatMoney(surplusValue, true)}. Remova esse valor
              dos títulos selecionados para solicitar sua antecipação.
            </>
          ) : (
            <>
              A antecipação foi cancelada, pois o valor líquido excedeu o limite mensal de
              antecipações definido pelo sacado.
              <br />
              <br />
              Aguarde o início do próximo mês ou a atualização dos limites.
            </>
          )}
        </Content>
        <ActionsRow>
          <Button onClick={handleOk}>OK</Button>
        </ActionsRow>
      </Container>
    </Modal>
  );
}

UnavailableLimitDialog.propTypes = {
  surplusValue: PropTypes.number,
  onLeaved: PropTypes.func,
  show: PropTypes.bool,
  redirectUrl: PropTypes.string,
};

UnavailableLimitDialog.defaultProps = {
  surplusValue: 0,
  onLeaved: () => {},
  show: false,
  redirectUrl: null,
};

export default UnavailableLimitDialog;
