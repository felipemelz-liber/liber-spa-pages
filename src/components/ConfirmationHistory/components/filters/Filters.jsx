import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Label, CleanAll, Filtered } from './Filters.styles.jsx';
import { DateFieldFilter } from '../../../../common-components/Filters/index.js';

const Filters = ({ count, handleFilter }) => {
  const [rangeField, setRangeField] = useState({
    start: null,
    end: null,
  });

  const handleClear = () => {
    const filters = {
      start: null,
      end: null,
    };
    setRangeField(filters);
    handleFilter(filters);
  };

  const handleDateFilter = rangeData => {
    setRangeField(rangeData);
    handleFilter(rangeData);
  };

  const { start, end } = rangeField;

  const renderOnFilter = () => {
    if (start || end) {
      return (
        <>
          <CleanAll onClick={handleClear}>Limpar filtros</CleanAll>
          <Filtered>{`${count} ${
            count === 1 ? 'confirmação filtrada' : 'confirmações filtradas'
          }`}</Filtered>
        </>
      );
    }
    return null;
  };

  return (
    <>
      <Label>Filtrar por</Label>
      <DateFieldFilter
        label="Data de Criação"
        description="Selecione uma data inicial e uma data final para o período de criação"
        onClear={handleClear}
        onFilter={handleDateFilter}
        selectedItems={[start, end]}
      />
      {renderOnFilter()}
    </>
  );
};

Filters.propTypes = {
  count: PropTypes.number.isRequired,
  handleFilter: PropTypes.func.isRequired,
};

export default Filters;
