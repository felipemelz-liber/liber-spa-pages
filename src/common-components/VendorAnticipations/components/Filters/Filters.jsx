import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import DateFieldFilter from '../../../Filters/DateFieldFilter/DateFieldFilter';
import { REQUESTED_LT, REQUESTED_GT, STATUS, STATUS_IN } from '../../utils';
import { updateFilter } from '../../actions/VendorAdvancesActions';
import OptionsFilter from '../../../Filters/OptionsFilter/OptionsFilter';
import { FilterText, CleanFilters } from './Filters.styles';
import { dispatchDataLayerEvent } from '../../../../vendor/Utils';

const clearDateFilters = {
  [REQUESTED_LT]: null,
  [REQUESTED_GT]: null,
};

const clearStatusFilters = {
  [STATUS_IN]: null,
};

const allClearFilters = {
  ...clearDateFilters,
  ...clearStatusFilters,
};

const onDateFilter = (changeFilters, applyedFilters, isAdmin) => range => {
  dispatchDataLayerEvent('antecipacoesFiltrarSolicitacao', isAdmin);

  let filters = {};

  if (range && range.end) {
    filters = {
      [REQUESTED_GT]: range.start.format(),
      [REQUESTED_LT]: range.end.format(),
    };
  } else {
    filters = {
      [REQUESTED_GT]: range && range.start ? range.start.startOf('day').format() : null,
      [REQUESTED_LT]: range && range.start ? range.start.endOf('day').format() : null,
    };
  }

  changeFilters({ ...applyedFilters, ...filters });
};

const onStatusFilter = (filters, changeFilters, selectedStatuses, isAdmin) => {
  const status = selectedStatuses
    .map(statusId => (statusId === STATUS.finished ? 'Concluida' : 'Cancelada'))
    .join(' | ');

  dispatchDataLayerEvent('antecipacoesFiltrarStatus', isAdmin, {
    filtroStatusAntecipacoes: status,
  });

  changeFilters({ ...filters, [STATUS_IN]: selectedStatuses });
};

const handleDateFilterClear = (changeFilters, filters, isAdmin) => {
  dispatchDataLayerEvent('antecipacoesLimparFiltro', isAdmin);

  changeFilters({ ...filters, ...clearDateFilters });
};

const handleStatusFilterClear = (changeFilters, filters, isAdmin) => {
  dispatchDataLayerEvent('antecipacoesLimparFiltro', isAdmin);

  changeFilters({ ...filters, ...clearStatusFilters });
};

const handleClearAllFiltersClick = (changeFilters, isAdmin) => {
  dispatchDataLayerEvent('antecipacoesLimparFiltro', isAdmin);

  changeFilters(allClearFilters);
};

const hasFilters = filters =>
  Object.keys(filters).reduce((flag, key) => flag || Boolean(filters[key]), false);

export const Filters = ({ changeFilters, filters, isAdmin }) => {
  return (
    <>
      <FilterText>Filtrar por</FilterText>
      <DateFieldFilter
        label="Solicitação"
        onFilter={onDateFilter(changeFilters, filters, isAdmin)}
        onClear={() => handleDateFilterClear(changeFilters, filters, clearDateFilters, isAdmin)}
        selectedItems={[filters[REQUESTED_GT], filters[REQUESTED_LT]]}
      />
      <OptionsFilter
        filterValues={filters[STATUS_IN]}
        onFilter={selectedStatuses =>
          onStatusFilter(filters, changeFilters, selectedStatuses, isAdmin)
        }
        onClear={() => handleStatusFilterClear(changeFilters, filters, clearStatusFilters, isAdmin)}
        options={[
          { option: 'Concluída', value: STATUS.finished },
          { option: 'Cancelada', value: STATUS.canceled },
        ]}
        label="Status"
      />
      {hasFilters(filters) ? (
        <CleanFilters onClick={() => handleClearAllFiltersClick(changeFilters, isAdmin)}>
          Limpar filtros
        </CleanFilters>
      ) : null}
    </>
  );
};

Filters.propTypes = {
  changeFilters: PropTypes.func,
  filters: PropTypes.shape({
    [REQUESTED_LT]: PropTypes.string,
    [REQUESTED_GT]: PropTypes.string,
    [STATUS_IN]: PropTypes.arrayOf(PropTypes.number),
  }),
  isAdmin: PropTypes.bool,
};

Filters.defaultProps = {
  filters: {},
  isAdmin: false,
  changeFilters: () => {},
};

const mapStateToProps = ({ vendorAdvances: { filters, isAdmin } }) => ({ filters, isAdmin });

const mapDispatchToProps = {
  changeFilters: updateFilter,
};

export default connect(mapStateToProps, mapDispatchToProps)(Filters);
