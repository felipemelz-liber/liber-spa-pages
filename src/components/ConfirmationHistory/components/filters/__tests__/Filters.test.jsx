import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';
import 'jest-styled-components';
import Filters from '../Filters';
import { DateFieldFilter } from '../../../../../common-components/Filters';

describe('Testing Filters Component', () => {
  configure({ adapter: new Adapter() });
  let props;
  beforeEach(() => {
    props = { count: 10, handleFilter: sinon.spy() };
  });
  it('should render correctly', () => {
    const wrapper = shallow(<Filters {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render correctly with calendar open', () => {
    const wrapper = shallow(<Filters {...props} />);
    const { handleFilter } = props;
    const tag = wrapper.find(DateFieldFilter);
    tag.simulate('filter', { start: '11/11/2001', end: '12/01/2002' });
    expect(handleFilter.calledWith({ start: '11/11/2001', end: '12/01/2002' })).toBeTruthy();
  });
  it('should call handleFilter on FilterButton click', () => {
    const wrapper = shallow(<Filters {...props} />);
    const filter = wrapper.find(DateFieldFilter);
    const { handleFilter } = props;
    filter.simulate('clear');
    expect(
      handleFilter.calledWith({
        start: null,
        end: null,
      }),
    ).toBeTruthy();
  });
});
