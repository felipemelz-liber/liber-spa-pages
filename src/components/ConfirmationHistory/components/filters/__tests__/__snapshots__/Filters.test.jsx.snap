// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Filters Component should render correctly 1`] = `
<Fragment>
  <styled.div>
    Filtrar por
  </styled.div>
  <DateFieldFilter
    description="Selecione uma data inicial e uma data final para o período de criação"
    label="Data de Criação"
    onClear={[Function]}
    onFilter={[Function]}
    selectedItems={
      Array [
        null,
        null,
      ]
    }
  />
</Fragment>
`;
